docs/design/specs/Ecma-335-Augments.md

```diff
@@ -810,3 +820,84 @@ Bullet 2: It is valid to call a virtual method using `call` (rather than `callvi
 the method is to be resolved using the class specified by method rather than as
 specified dynamically from the object being invoked. This is used, for example, to
 compile calls to “methods on `base`” (i.e., the statically known parent class) or to virtual static methods.
+
+## Covariant Return Types
+
+Covariant return methods is a runtime feature designed to support the [covariant return types](https://github.com/dotnet/csharplang/blob/master/proposals/csharp-9.0/covariant-returns.md) and [records](https://github.com/dotnet/csharplang/blob/master/proposals/csharp-9.0/records.md) C# language features implemented in C# 9.0.
+
+This feature allows an overriding method to have a return type that is different than the one on the method it overrides, but compatible with it. The type compability rules are defined in ECMA I.8.7.1.
+
+Example:
+```
+class Base
+{
+  public virtual object VirtualMethod() { return null; }
+}
+
+class Derived : Base
+{
+  // Note that the VirtualMethod here overrides the VirtualMethod defined on Base even though the 
+  // return types are not the same. However, the return types are compatible in a covariant fashion.
+  public override string VirtualMethod() { return null;}
+}
+```
+
+Covariant return methods can only be described through MethodImpl records, and are only be applicable to methods on non-interface reference types. In addition the covariant override is only applicable for overriding methods also defined on a non-interface reference type.
```

Comment:
One last: "are only be applicable" -> "are only applicable"

Reply:
Nice, I missed that one :D.